db/import.go

```diff
@@ -0,0 +1,193 @@
+		newRev = createRevID(generation, parentRev, body)
+		base.LogTo("Import", "Created new rev ID %v", newRev)
+		body["_rev"] = newRev
+		doc.History.addRevision(docid, RevInfo{ID: newRev, Parent: parentRev, Deleted: isDelete})
+
+		// During import, oldDoc (doc.Body) is nil (since it's no longer available)
+		doc.body = nil
+
+		// Note - no attachments processing is done during ImportDoc.  We don't (currently) support writing attachments through anything but SG.
+
+		return body, nil, nil
+	})
+
+	switch err {
+	case base.ErrAlreadyImported:
+		// If the doc was already imported, we want to return the imported version
+		docOut = alreadyImportedDoc
+	case nil:
+		base.LogTo("Import+", "Imported %s (delete=%v) as rev %s", docid, isDelete, newRev)
+	case base.ErrImportCancelled:
+		// Import was cancelled (SG purge) - don't return error.
+	case base.ErrImportCancelledFilter:
+		// Import was cancelled based on import filter.  Return error but don't log as error/warning.
+		return nil, err
+	case base.ErrImportCasFailure:
+		// Import was cancelled due to CAS failure.
+		return nil, err
+	default:
+		base.LogTo("Import", "Error importing doc %q: %v", docid, err)
+		return nil, err
+
+	}
+
+	return docOut, nil
+}
+
+//////// Import Filter Function
+
+// A compiled JavaScript event function.
+type jsImportFilterRunner struct {
+	sgbucket.JSRunner
+	response bool
+}
+
+// Compiles a JavaScript event function to a jsImportFilterRunner object.
+func newImportFilterRunner(funcSource string) (sgbucket.JSServerTask, error) {
+	importFilterRunner := &jsEventTask{}
+	err := importFilterRunner.Init(funcSource)
+	if err != nil {
+		return nil, err
+	}
+
+	importFilterRunner.After = func(result otto.Value, err error) (interface{}, error) {
+		nativeValue, _ := result.Export()
+		return nativeValue, err
+	}
+
+	return importFilterRunner, nil
+}
+
+type ImportFilterFunction struct {
+	*sgbucket.JSServer
+}
+
+func NewImportFilterFunction(fnSource string) *ImportFilterFunction {
+
+	base.LogTo("Import+", "Creating new ImportFilterFunction")
+	return &ImportFilterFunction{
+		JSServer: sgbucket.NewJSServer(fnSource, kTaskCacheSize,
+			func(fnSource string) (sgbucket.JSServerTask, error) {
+				return newImportFilterRunner(fnSource)
+			}),
+	}
+}
+
+// Calls a jsEventFunction returning an interface{}
+func (i *ImportFilterFunction) EvaluateFunction(doc Body) (bool, error) {
+
+	result, err := i.Call(doc)
+	if err != nil {
+		base.Warn("Unexpected error invoking import filter for document %s - processing aborted, document will not be imported.  Error: %v", err)
+		return false, err
+	}
+	switch result := result.(type) {
+	case bool:
+		return result, nil
+	case string:
+		boolResult, err := strconv.ParseBool(result)
+		if err != nil {
+			return false, err
+		}
+		return boolResult, nil
+	default:
+		base.Warn("Import filter function returned non-boolean result %v", result)
+		return false, errors.New("Import filter function returned non-boolean value.")
```

Comment:
How about changing this to:

```
base.Warn("Import filter function returned non-boolean result %v.  Type: %T", result, result)
```

Might save a round of debugging to figure out what type was returned.